src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1398,6 +1455,104 @@ private MqttServerStream resolvePublisher(
             return stream;
         }
 
+        private void resolveSession(
+            long traceId,
+            long authorization,
+            int flags,
+            int willDelay,
+            byte willQos,
+            byte willRetain,
+            int expiryInterval,
+            String16FW contentType,
+            MqttPayloadFormat payloadFormat,
+            String16FW responseTopic,
+            OctetsFW correlationData,
+            OctetsFW willPayload)
+        {
+            final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
+            final int topicKey = topicKey(topic);
+
+            final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
+
+            MqttSessionStateStream stream = sessionStreams.computeIfAbsent(topicKey, s -> new MqttSessionStateStream(0, topic));
+            if (route != null)
+            {
+                final long resolvedId = route.correlationId();
+                stream.initRoute(resolvedId);
+
+                final JsonObject sessionObj = objectBuilder.add(SESSION_EXPIRES_AT_NAME, sessionExpiresAt)
+                                                           .add(WILL_DELAY_NAME, willDelay)
+                                                           .build();
```

Comment:
Does it make sense to include `willDelay: 0` if there is no `will` message?

Also, given that the session expires a N seconds after the client disconnects abruptly, this `sessionExpiresAt` value should be updated periodically to make it maximally accurate when the client attempts to reconnect.